tests/compliance.js

```diff
@@ -245,7 +245,7 @@ function test_selection_compliance (SelectionClass, setup=function () {}, extra_
             // Not all Selections might use the Xapian engine, but I can't think
             // of a better place to put this
             let engine = MockEngine.mock_default();
-            engine.query_finish.and.returnValue({ models: [] });
+            engine.query_promise.and.returnValue(Promise.resolve({ models: [] }));
```

Comment:
Did that at first, fully embracing promises made for less code in mock engine so I preferred it. And the nice thing here is you can easily bounce between Promise.resolve and Promise.reject for the simple cases, as well as actually supporting more of an async flow (waiting some ticks on the mainloop e.g.) if you wanted to test application code with a real delay for some reason...